fvm/script.go

```diff
@@ -24,7 +24,7 @@ type ScriptProcedure struct {
 }
 
 func Script(code []byte) *ScriptProcedure {
-	scriptHash := hash.DefaultHasher.ComputeHash(code)
+	scriptHash := hash.DefaultComputeHash(code)
```

Comment:
Not a comment on this PR:

In my mind the ID of the script also contained the script arguments... Its interesting that it doesn't. As a consequence a script like a get balance script that accepts an address as an argument will always have the same ID no mater which account it queries. Is this problematic for the random generation/usage?

Reply:
Good point!
I don't know if some applications require randoms to be diversified per script argument. There may be some application with such requirement, so I added this: https://github.com/onflow/flow-go/pull/4648/commits/a00d5f0204d4c08afb886417b31576cb9a9d7b02

However I didn't update the other `ScriptProcedure` ID. I don't know where that is used.